### PR TITLE
Join event as interested

### PR DESCRIPTION
**Pull request by MikaelKalajdzic**
This PR introduces the feature of joining an event as an interested participant instead of joining it directly as it is currently.
This allows users to show interest to some activities so that the organizer can have an idea of the potential participants, but still give flexibility to users to really join or not at the end.

### CODE REVIEW

File: app/src/main/java/com/sdpteam/connectout/event/Event.java
Comment (TheAmericanMomo) [inline app/src/main/java/com/sdpteam/connectout/event/Event.java:1]: Nice! The logic adapts rather well with what we already had for event registration. It is also compatible with the changes that I made for the normal event registration. Great job. The merge shouldn't be too much of a pain

File: app/src/main/java/com/sdpteam/connectout/event/creator/EventCreatorActivity.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ private void saveEvent(String title, GPSCoordinates coordinates, String descript
 
         //Create associated event.
         //TODO add yourself to the participants by default?
-        Event newEvent = new Event(eventCreatorViewModel.getUniqueId(), title, description, coordinates, ownerId, null, date);
+        Event newEvent = new Event(eventCreatorViewModel.getUniqueId(), title, description, coordinates, ownerId);
```
Comment (TheAmericanMomo) [inline app/src/main/java/com/sdpteam/connectout/event/creator/EventCreatorActivity.java:93]: I dealt with this TODO in my PR so no worries

File: app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/filter/EventParticipantIdFilter.java [hunk 2]
```diff
@@ -19,6 +19,6 @@ public boolean test(Event event) {
         if (userId == null) {
             return false;
         }
-        return userId.equals(event.getOrganizer()) || event.getParticipants().contains(userId);
+        return userId.equals(event.getOrganizer()) || event.hasJoined(userId);
```
Comment (TheAmericanMomo) [inline app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/filter/EventParticipantIdFilter.java:22]: Interesting. So are you saying that a user that has "joined" should be considered in the filtering? please enlighten me on why this should be the case

File: app/src/main/java/com/sdpteam/connectout/event/viewer/EventActivity.java [hunk 3]
```diff
@@ -78,32 +80,48 @@ private void initViewModel() {
...
         TextView description = findViewById(R.id.event_description);
-        Button participationBtn = findViewById(R.id.event_join_button);
+        Button joinBtn = findViewById(R.id.event_join_button);
+        Button interestedBtn = findViewById(R.id.event_interested_button);
         Button participantsBtn = findViewById(R.id.event_participants_button);
         ImageButton chatBtn = findViewById(R.id.event_chat_btn);
 
-        participationBtn.setOnClickListener(v -> {
-            eventViewModel.toggleParticipation(currentUserId);
-        });
```
Comment (TheAmericanMomo) [inline app/src/main/java/com/sdpteam/connectout/event/viewer/EventActivity.java:87]: not sure why you delete this. Anyways, it's an essential part of my pr so I ll see that with you when we meet up for the merge

File: app/src/main/java/com/sdpteam/connectout/event/viewer/EventActivity.java
Comment (TheAmericanMomo) [inline app/src/main/java/com/sdpteam/connectout/event/viewer/EventActivity.java:1]: The logic works, we re just going to have to merge it with the code that I wrote. Might be very tricky

File: app/src/main/java/com/sdpteam/connectout/event/viewer/EventViewModel.java
Comment (TheAmericanMomo) [inline app/src/main/java/com/sdpteam/connectout/event/viewer/EventViewModel.java:1]: same here, I made completely different changes in this class, we ll figure out a solution when we meet up.

**Review by TheAmericanMomo — APPROVED** [review]
Ok so I like the implementation that you made and the UI is nice. I'm not requesting any changes, because we ll make them when we merge. I'm using quite a few functions that you deleted so we ll see how that goes.

File: app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/filter/EventParticipantIdFilter.java (hunk 2 above)
Comment (MikaelKalajdzic) [reply app/src/main/java/com/sdpteam/connectout/event/nearbyEvents/filter/EventParticipantIdFilter.java:22]: It does the same, it's just refactoring

**Pull request merged**